src/main.c

```diff
@@ -21,14 +23,16 @@ int	main(int argc, char **argv)
 
 	if (argc != 2)
 	{
-		printf("Error\n");
-		exit(123);
+		ft_perror("Usage: ./miniRT file.rt", 0);
+		exit(1);
 	}
-	ft_create_scene(&scene);
+	if (ft_parse_file(argv[1], &scene, &cam))
+		exit(1);
 	ft_initiate_image(&image);
 	ft_initiate_camera(&cam);
 	ft_initiate_viewport(&viewport, cam, image);
 	ft_create_image(image, cam, viewport, scene);
-	(void)argv;
+	free(scene.obj);
```

Comment:
Could summarise these two in one function ft_free_all or ft_shutdown so it looks less bare metal

Reply:
Was just temporary to check for leaks with valgrind. I would leave it like that until we know what we really need to clean at the end

Reply:
I don't think there will be anything else. These are the only two instance requiring mallocing

Reply:
I saw a camera implementation which caches the ray directions - this cache would also need to be freed.
Another malloc would be for mlx. Though this will be contained in the render part I think.
What I want to say: you'll never know whats coming.

But yeah I can move it in its own function